analyzers/tests/SonarAnalyzer.UnitTest/Rules/SingleStatementPerLineTest.cs

```diff
@@ -40,6 +42,21 @@ public class SingleStatementPerLineTest
                 .WithTopLevelStatements()
                 .Verify();
 
+        [TestMethod]
+        public void SingleStatementPerLine_Razor() =>
+            builderCS.AddSnippet(
+"""
+@if (true) { <p>Test</p> } <!-- Compliant FN -->
```

Comment:
I am not sure this is actually a FN, since the body of the if is HTML and not really another C# statement.
That is up for a debate, what really should not be up for a debate is if you would inset some C# statement inside the if body.

Reply:
Good point! I just added a nested [@if conditional control](https://learn.microsoft.com/en-us/aspnet/core/mvc/views/razor?view=aspnetcore-7.0#conditionals-if-else-if-else-and-switch)